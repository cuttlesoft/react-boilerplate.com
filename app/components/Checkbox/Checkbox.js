/* eslint-disable react/prop-types */
/**
 *
 * Chexbox
 *
 */

import React from 'react'
import { CheckBox } from 'grommet'
import { doc } from './Checkbox.doc'
import helpers from '../../utils/helpers'

function createCheckbox({ checked, disabled, id, label, name, onChange, reverse, toggle }) {
  return (
    <CheckBox
      checked={checked}
      disabled={disabled}
      id={id}
      label={label}
      name={name}
      onChange={onChange}
      reverse={reverse}
      toggle={toggle}
    />
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createCheckbox,
})
