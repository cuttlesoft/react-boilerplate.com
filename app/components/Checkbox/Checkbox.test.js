import React from 'react'
import { render } from '@testing-library/react'

import Checkbox from './Checkbox'

const renderComponent = props => render(<Checkbox label="Choice" {...props} />)

/**
 *
 * Tests for Chexbox
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
describe('Checkbox', () => {
  it('renders and matches snapshot', () => {
    const props = {
      checked: true,
      disabled: false,
      id: 'checkbox-1',
      label: 'New Label',
      name: 'Some name',
      onChange: () => {},
      reverse: true,
      toggle: true,
    }
    const { container } = renderComponent(props)

    expect(container).toMatchSnapshot()
  })
})
