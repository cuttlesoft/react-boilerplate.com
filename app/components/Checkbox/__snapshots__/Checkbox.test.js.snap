// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders and matches snapshot 1`] = `
<div>
  <label
    class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
    for="checkbox-1"
  >
    <span>
      New Label
    </span>
    <div
      class="StyledBox-sc-13pk1d4-0 hXzPDn StyledCheckBox-sc-1dbk5ju-6 CBoMo"
    >
      <input
        checked=""
        class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 ljjPfS"
        id="checkbox-1"
        name="Some name"
        type="checkbox"
      />
      <span
        class="StyledCheckBox__StyledCheckBoxToggle-sc-1dbk5ju-4 hqYniG"
      >
        <span
          class="StyledCheckBox__StyledCheckBoxKnob-sc-1dbk5ju-5 bVGQgu"
        />
      </span>
    </div>
  </label>
</div>
`;
